client/starwhale/api/_impl/data_store.py

```diff
@@ -1128,6 +1119,23 @@ def update_table(
             else:
                 table.insert(r)
 
+    def _get_table(self, table_name: str, schema: TableSchema) -> MemoryTable:
+        table = self.tables.get(table_name, None)
```

Comment:
remove the duplicated condition check? make all get table operations in the lock context?

Reply:
The reason to use double check is to avoid waits for tables which have different names that has already completed initialization,  but i'm not sure that double-checking yields this benefit

Reply:
done